plugins/samson_ledger/test/lib/samson_ledger/client_test.rb

```diff
@@ -29,10 +40,71 @@
   end
 
   describe ".post_deployment" do
+    before do
+      @asserts = Proc.new {}
```

Comment:
@grosser, did some more digging into this. It's getting bizarre..

So there are 4 requirements for the random segfaults to happen randomly:
1. `require 'single_cov'`
2. Add my new `#create_singleton_struct` method inside `client_test.rb` instead of `support/struct_helper.rb`
3. Use `let(:results) { [] }` and add item to it inside the `request_lambda` (instance variable `@results = []` does not cause the crash to happen)
4. Do an AR write in the test - `prod_env = environments(:production)`

Take out any of the above will make the random segfaults go away. 

This is more informational at this point. So I now use `let(:results) { [] }` and puts `#create_singleton_struct` in `support/struct_helper.rb`. No crashes.

Github review UI is bugging out like you said. This view is still showing the old code even after I rebased and pushed. Please review when you have a moment. Thanks!